src/analysis/special_functions/gamma.lean

```diff
@@ -519,3 +534,425 @@ end
+    conv in (𝓝 _) { rw ←add_zero (1 : ℝ) },
+    exact tendsto_const_nhds.add (tendsto_const_nhds.div_at_top tendsto_coe_nat_at_top_at_top) },
+end
+
+variables {f : ℝ → ℝ} {x : ℝ} {n : ℕ}
+
+lemma f_nat_eq (hf_feq : ∀ {y:ℝ}, 0 < y → f (y + 1) = f y + log y) (hn : n ≠ 0) :
+  f n = f 1 + log (nat.factorial (n - 1)) :=
+begin
+  induction n with n h_ind,
+  { contrapose! hn, tauto },
+  rcases nat.eq_zero_or_pos n with rfl|hn',
+  { simp },
+  { rw [nat.cast_succ, hf_feq (nat.cast_pos.mpr hn'), h_ind hn'.ne'],
+    have : n.succ - 1 = (n - 1).succ,
+    { simp_rw nat.succ_eq_add_one,
+      rw [nat.add_sub_cancel, nat.sub_add_cancel],
+      linarith },
+    rw [this, nat.factorial_succ, nat.cast_mul,
+      log_mul _ (nat.cast_ne_zero.mpr (nat.factorial_ne_zero _)),
+      nat.sub_add_cancel (by linarith : 1 ≤ n)],
+    { ring },
+    { rw [nat.cast_ne_zero, nat.sub_add_cancel (by linarith : 1 ≤ n)],
+      exact hn'.ne', } },
+end
+
+open_locale big_operators
+
+lemma f_add_nat_eq (hf_feq : ∀ {y:ℝ}, 0 < y → f (y + 1) = f y + log y) (hx : 0 < x) (n : ℕ) :
+  f (x + n) = f x + ∑ (m : ℕ) in finset.range n, log (x + m) :=
+begin
+  induction n with n hn,
+  { simp },
+  { have : x + n.succ = (x + n) + 1,
+    { push_cast, ring },
+    rw [this, hf_feq, hn],
+    rw [finset.range_succ, finset.sum_insert (finset.not_mem_range_self)],
+    abel,
+    linarith [(nat.cast_nonneg n : 0 ≤ (n:ℝ))] },
+end
+
+/-- Linear upper bound for `f (x + n)` on unit interval -/
+lemma f_add_nat_le
+  (hf_conv : convex_on ℝ (Ioi 0) f) (hf_feq : ∀ {y:ℝ}, 0 < y → f (y + 1) = f y + log y)
+  (hn : n ≠ 0) (hx : 0 < x) (hx' : x ≤ 1) :
+  f (n + x) ≤ f n + x * log n :=
+begin
+  have hn': 0 < (n:ℝ) := nat.cast_pos.mpr (nat.pos_of_ne_zero hn),
+  have : f n + x * log n = (1 - x) * f n + x * f (n + 1),
+  { rw [hf_feq hn'], ring, },
+  rw [this, (by ring : (n:ℝ) + x = (1 - x) * n + x * (n + 1))],
+  simpa only [smul_eq_mul] using hf_conv.2 hn' (by linarith : 0 < (n + 1 : ℝ))
+    (by linarith : 0 ≤ 1 - x) hx.le (by linarith),
+end
+
+/-- Linear lower bound for `f (x + n)` on unit interval -/
+lemma f_add_nat_ge
+  (hf_conv : convex_on ℝ (Ioi 0) f) (hf_feq : ∀ {y:ℝ}, 0 < y → f (y + 1) = f y + log y)
+  (hn : 2 ≤ n) (hx : 0 < x) :
+  f n + x * log (n - 1) ≤ f (n + x) :=
+begin
+  have npos : 0 < (n:ℝ) - 1,
+  { rw [←nat.cast_one, sub_pos, nat.cast_lt], linarith, },
+  have c := (convex_on_iff_slope_mono_adjacent.mp $ hf_conv).2
+    npos (by linarith : 0 < (n:ℝ) + x) (by linarith : (n:ℝ) - 1 < (n:ℝ)) (by linarith),
+  rw [add_sub_cancel', sub_sub_cancel, div_one] at c,
+  have : f (↑n - 1) = f n - log (↑n - 1),
+  { nth_rewrite_rhs 0 (by ring : (n:ℝ) = (↑n - 1) + 1),
+    rw [hf_feq npos, add_sub_cancel] },
+  rwa [this, le_div_iff hx, sub_sub_cancel, le_sub_iff_add_le, mul_comm _ x, add_comm] at c,
+end
+
+variables (n x)
+
+/-- The function `n ↦ x log n + log n! - (log x + ... + log (x + n))`, which tends to `log Γ(x)` as
+`n → ∞`. -/
+def log_gamma_seq : ℝ :=
```

Comment:
can you put explicitely the variables in the definition, so that the reader can follow more easily what is going on?